### PR TITLE
Rollforward 2.6 changes 2.7

### PR DESCRIPTION
**Pull request by ibsoln**
This PR comprises cherry-picked 'roll-forward' commits from 2.6 to 2.7

2.7 was branched from 2.6 at 14:19 on 28 November (by @jamiltz ) since that time these commits were applied to 2.6 and are now brought into 2.7:

- Commit: 494fbfee89d4ca7aecfcd0eac9f6625e169df282
        DOC-4912 - upgrade scenario edits (#119)

- Commit: c399e78df6cbcd9d6bdabc458e38037d7852d051
    DOC-5782 - replace swagger spec links (#117)

- Commit: ef53628c496c1d06714ab36b8b85ae0cfae15db1
    DOC-5902 - default revs_limit value (#115)
    Updates to sg.yaml, relnotes

- Commit: 72bf78cdeaf410dac832430b19808ff5c78982c2
    DOC-5872 (#112)

- Commit: 0578d1be8a366d6f3dac648c3345f86192023246 
	Clarity on exposedFeatures configuration requirement (#145) 
	* Update deploy-cluster.adoc

- Commit:  f8de90a43989d0efbada637f8f0dd8618dcd3bb7
    DOC-5711 - current docs cleanup (#111)  
    OUTCOME-updates to nav, sg.yaml and shared_bucket_access

- Commit:  98edda42afe525b0434d7055d208a8a2f1331d43
    DOC-5352-Case-Change (#123)

Commit: f121600c0f2793bfccd6eb884f2366e64214caf7
    terminology improvement (#142)

### CODE REVIEW

File: modules/ROOT/assets/attachments/sg.yaml [hunk 1]
```diff
@@ -312,43 +314,28 @@ properties:
           enable_shared_bucket_access:
             type: boolean
             description: |
-              Introduced in Sync Gateway 1.5, this property specifies whether to enable Mobile-Web Data Sync and works in conjunction with the [import_docs](#databases-foo_db-import_docs) property.
+              Introduced in Sync Gateway 1.5, this property specifies whether to enable Mobile-Server Data Sync (a.k.a _mobile convergence_). You can learn more about this functionality [here](https://docs.couchbase.com/sync-gateway/current/shared-bucket-access.html)
```
Comment (RichardSmedley) [inline modules/ROOT/assets/attachments/sg.yaml:317]: This is an MD style link
For asciidoc needs xref:sync-gateway::shared-bucket-access.adoc[here]

**Comment by ibsoln** [comment]
@rajagp / @couchbase/docs-committers 
Compiled this roll-forward PR as best I know. Needs a sanity check I'd say.
Today I learned ... do these darned things as you go along.

File: modules/ROOT/assets/attachments/sg.yaml [hunk 2]
```diff
@@ -312,43 +314,28 @@ properties:
...
+              Users should set the server's metadata purge interval based on their expected client replication frequency, to ensure that clients are notified of the tombstone prior to that tombstone being purged.
 
-              The default Metadata Purge Interval is set to 3 days which can potentially result in tombstones being purged before all clients have had a chance to get notified of it.
+              NOTE: The default Metadata Purge Interval is set to 3 days which can potentially result in tombstones being purged before all clients have had a chance to get notified of it.
 
               Ways to tune the Metadata Purge Interval on Couchbase Server:
 
-              - Bucket settings [on UI](https://developer.couchbase.com/documentation/server/5.0/settings/configure-compact-settings.html)
-              - Bucket endpoint [on the REST API](https://developer.couchbase.com/documentation/server/4.6/rest-api/rest-bucket-create.html)
+              - Bucket settings [on UI](https://developer.couchbase.com/documentation/server/current/settings/configure-compact-settings.html)
```
Comment (RichardSmedley) [inline modules/ROOT/assets/attachments/sg.yaml:337]: See above note on xref

File: modules/ROOT/assets/attachments/sg.yaml [hunk 3]
```diff
@@ -312,43 +314,28 @@ properties:
...
+              - Bucket endpoint [on the REST API](https://developer.couchbase.com/documentation/server/current/rest-api/rest-bucket-create.html)
```
Comment (RichardSmedley) [inline modules/ROOT/assets/attachments/sg.yaml:338]: ditto

File: modules/ROOT/assets/attachments/sg.yaml (hunk 1 above)
Comment (ibsoln) [reply modules/ROOT/assets/attachments/sg.yaml:317]: I thought yaml was MD ... isn't that different

**Review by RichardSmedley — COMMENTED** [review]
Content looks fine to me,
but see comments on xref
and I also suggest using asciidoc style links in shared-bucket-access.adoc 
Cheers :+1:

File: modules/ROOT/assets/attachments/sg.yaml (hunk 1 above)
Comment (RichardSmedley) [reply modules/ROOT/assets/attachments/sg.yaml:317]: Ah, my bad, I totally disregarded the file extension :face-palm:

Comment (ibsoln) [reply modules/ROOT/assets/attachments/sg.yaml:317]: np

File: modules/ROOT/assets/attachments/sg.yaml (hunk 2 above)
Comment (RichardSmedley) [reply modules/ROOT/assets/attachments/sg.yaml:337]: YAML, ignore above...

File: modules/ROOT/assets/attachments/sg.yaml (hunk 3 above)
Comment (RichardSmedley) [reply modules/ROOT/assets/attachments/sg.yaml:338]: et encore une fois :^)

**Review by RichardSmedley — COMMENTED** [review]
> and I also suggest using asciidoc style links in shared-bucket-access.adoc

So just that, then :)

**Review by ibsoln — COMMENTED** [review]
> > and I also suggest using asciidoc style links in shared-bucket-access.adoc
> 
> So just that, then :)
Thanks @RichardSmedley ... 

I may be misremembering @jamiltz, but I think when you're linking into that sg.yamll file you can't use xref. The file is pre-processed or something by ?swagger? and until that point the thing you are linking to (the anchor) doesn't exist. May xref cannot cope with it? Maybe .yaml is a mobile thing only?

**Comment by RichardSmedley** [comment]
> > > and I also suggest using asciidoc style links in shared-bucket-access.adoc
> > 
> > 
> > So just that, then :)
> > Thanks @RichardSmedley ...
> 
> I may be misremembering @jamiltz, but I think when you're linking into that sg.yamll file you can't use xref. The file is pre-processed or something by ?swagger? and until that point the thing you are linking to (the anchor) doesn't exist. May xref cannot cope with it? Maybe .yaml is a mobile thing only?

Ah, possibly - maybe I need to avoid commenting and reviewing so far out of my areas of knowledge 
- there's none of this used in the SDK docs. :)
Pleased to have learned something new at any rate :+1:

**Comment by ibsoln** [comment]
> > > > and I also suggest using asciidoc style links in shared-bucket-access.adoc
> > > 
> > > 
> > > So just that, then :)
> > > Thanks @RichardSmedley ...
> > 
> > 
> > I may be misremembering @jamiltz, but I think when you're linking into that sg.yamll file you can't use xref. The file is pre-processed or something by ?swagger? and until that point the thing you are linking to (the anchor) doesn't exist. May xref cannot cope with it? Maybe .yaml is a mobile thing only?
> 
> Ah, possibly - maybe I need to avoid commenting and reviewing so far out of my areas of knowledge
> 
> * there's none of this used in the SDK docs. :)
>   Pleased to have learned something new at any rate 👍

No, don't stop! I might be totally wrong, but it sounds plausible. Whatever gets 'generated' is stuffed into config.properties.adoc using this macro `json_config_ui`

````
== Configuration Reference

json_config_ui::{attachmentsdir}/sg.yaml[]
````

Andi it's within that bit that the anchors we want ot link-to reside.
This kind of shows what's happeneng https://developer.ibm.com/articles/wa-use-swagger-to-document-and-define-restful-apis/#converting-a-yaml-document-into-an-html-page

**Comment by ibsoln** [comment]
@rajagp - do these changes look ok?

**Comment by simon-dew** [comment]
> I may be misremembering @jamiltz, but I think when you're linking into that sg.yamll file you can't use xref. The file is pre-processed or something by ?swagger? and until that point the thing you are linking to (the anchor) doesn't exist. May xref cannot cope with it? Maybe .yaml is a mobile thing only?

YAML (the file format) is agnostic about any markup that you use for text formatting. In [Swagger][1] / [OpenAPI][2], it's conventional to use markdown for text markup, but some tools also support AsciiDoc — e.g. [Swagger2Markup][3], which is what I use for the REST API documentation.

I haven't looked closely at the tools this project uses to process sg.yaml, so I can't comment on this specific instance. HTH 😉 

[1]: https://swagger.io/
[2]: https://www.openapis.org
[3]: https://github.com/Swagger2Markup/swagger2markup

**Review by amarantha-k — APPROVED** [review]
lgtm

**Pull request merged**